src/Numeric/Units/Dimensional/Variants.hs

```diff
@@ -68,9 +69,21 @@ infixl 7  *
 -- The product of quantities is a quantity.
 type family (v1 :: Variant) * (v2 :: Variant) :: Variant where
   'DUnit m1  * 'DUnit m2  = 'DUnit 'NonMetric
-  'DQuantity * 'DQuantity = 'DQuantity 
+  'DQuantity s1 * 'DQuantity s2 = 'DQuantity (s1 E.* s2)
```

Comment:
This is a problem. Prior to fixed point, the types of multiplication and division both used the variant product, because they did the same thing. Now that quantities carry a scale factor, this doesn't work (even though it still typechecks) because the scale factor on the result won't be correct.